plugin/src/main/java/com/denizenscript/denizen/objects/PlayerTag.java

```diff
@@ -2617,6 +2617,39 @@ else if (foodLevel / maxHunger < 1) {
             NMSHandler.playerHelper.refreshPlayer(object.getPlayerEntity());
         });
 
+        // <--[mechanism]
+        // @object PlayerTag
+        // @name stop_sound
+        // @input ElementTag
+        // @description
+        // Stops all sounds of the specified type for the player.
+        // Valid types are AMBIENT, BLOCKS, HOSTILE, MASTER, MUSIC, NEUTRAL, PLAYERS, RECORDS, VOICE, and WEATHER
+        // Instead of a type, you can specify a sound <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Sound.html>
+        // or a full sound key, which usually has the 'minecraft:' prefix.
+        // If no sound type is specified, all types will be stopped.
+        // -->
+        registerOnlineOnlyMechanism("stop_sound", (object, mechanism) -> {
+            SoundCategory category = null;
+            String key = null;
+            if (mechanism.hasValue()) {
+                if (mechanism.getValue().matchesEnum(SoundCategory.class)) {
+                    category = mechanism.getValue().asEnum(SoundCategory.class);
+                    if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
+                        object.getPlayerEntity().stopSound(category);
+                        return;
+                    }
+                }
+                else if (mechanism.getValue().matchesEnum(Sound.class) && NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {

```

Comment:
this check is backwards - must validate NMS version -before- doing anything with version locked features. Please also test on a 1.18 server to make sure nothing breaks